src/main/webapp/dev/js/common/datepicker.js

```diff
@@ -73,11 +75,21 @@ function prepareDatepickers() {
         showOtherMonths: true,
         gotoCurrent: true,
         defaultDate: yesterday,
-        maxDate: today,
+        maxDate: defaultStartDate,
         onSelect() {
             const newPublishDate = getMinDateForPublishDate($('#visibledate').datepicker('getDate'));
             $('#publishdate').datepicker('option', 'minDate', newPublishDate);
         },
+        onClose() {
+            if ($('#startdate').val() === $('#visibledate').val()) {
+                const $startTime = $('#starttime');
+                const $visibleTime = $('#visibletime');
+                // the visible time should be before the start time
```

Comment:
As the two can be equal, it'd be better to say `the visible time should not be later than the start time`.